### PR TITLE
feat: track archived ports

### LINKED ISSUE #2409: Track archived repositories in `ports.yml`
## Description

The number of archived repositories returned by the GitHub API isn't so accurate as we archived and migrated all existing userstyle repositories last year. 

IMO the only _real_ archived repositories are:
- [catppuccin/steam](https://github.com/catppuccin/steam)
- [catppuccin/aliucord](https://github.com/catppuccin/aliucord)
- https://github.com/catppuccin/sidebery

I think this distinction is important is because we may be open to unarchiving the above repositories in the future, but the userstyle repositories will not be unarchived as they now exist in https://github.com/catppuccin/userstyles.

I'd like to put these 3 repositories on the website and explain our reasoning for archiving and (optionally) the conditions it needs to meet for the repository to be unarchived. The reasoning can be embedded into the `.yml` file.

## Acceptance Criteria

- `ports.schema.json` is updated. I think it's nicer if we have a separate `archived` root level key with a list of `port`s but maybe it's nicer to have an `archived` key for each `port`.
- `ports.yml` is updated listing the archived repositories mentioned above.

### PR DESCRIPTION
**Pull request by sgoudham**
Closes #2409 

Outstanding questions:

- Do we want to display the archived repositories on the README in this repository? The future goal is to point everyone to our website so I didn't bother touching the deno code.
- I thought about adding an `archive_kind` enum but just opted for a simple string to track the reasoning, maybe worth revisiting.

### CODE REVIEW

File: resources/ports.yml [hunk 1]
```diff
@@ -2239,4 +2241,4 @@ showcases:
   - title: Reciper
     description: A simple, open-source recipe management app.
     link: https://github.com/judemont/reciper
-# yaml-language-server: $schema=https://raw.githubusercontent.com/catppuccin/catppuccin/main/resources/ports.schema.json
+# yaml-language-server: $schema=ports.schema.json
```
Comment (backwardspy) [inline resources/ports.yml:2244]: is it better to keep this the way it was for the (admittedly small) case of editing this file outside of the context of this repo?

**Review by backwardspy — COMMENTED** [review]
just the one comment

> Do we want to display the archived repositories on the README in this repository? The future goal is to point everyone to our website so I didn't bother touching the deno code.

i was also thinking about this, i too think we should just push for these to be tracked on the website and move away from relying on the readme.

> I thought about adding an archive_kind enum but just opted for a simple string to track the reasoning, maybe worth revisiting.

this seems fine, there are so many possible reasons for archival that free-form text beats an enum every time imo.

File: resources/ports.yml (hunk 1 above)
Comment (sgoudham) [reply resources/ports.yml:2244]: This is a regression, thanks for catching it 😄 

I change it locally when I need to iterate fast but _always_ somehow forget to change it back after I'm finished.

Comment (sgoudham) [inline resources/ports.yml:2244]:
```suggestion
# yaml-language-server: $schema=https://raw.githubusercontent.com/catppuccin/catppuccin/main/resources/ports.schema.json
```

**Comment by sgoudham** [comment]
> i was also thinking about this, i too think we should just push for these to be tracked on the website and move away from relying on the readme.
> 
> this seems fine, there are so many possible reasons for archival that free-form text beats an enum every time imo.

Agree with both statements :saluting_face:

**Pull request merged**